### PR TITLE
OpenCL: add add, subtract, multiply prim & rev support for the (scalar, matrix) case

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

Adds OpenCL support for prim and rev for add, subtract, multiply for cases where one of the arguments is a scalar.

## Tests

/

## Side Effects

/

## Release notes

Added OpenCL support for prim and rev for add, subtract, multiply for cases where one of the arguments is a scalar.

## Checklist

- [x] Math issue #1854 

- [ ] Copyright holder: Rok Češnovar, Uni. of Ljubljana

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

File: stan/math/opencl/rev/add.hpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  reverse_pass_callback([a_arena, b_arena, res]() mutable {
+    if (!is_constant<T_a>::value) {
+      auto& a_adj = forward_as<var_value<matrix_cl<double>>>(a_arena).adj();
+      a_adj = a_adj + res.adj();
+    }
+    if (!is_constant<T_b>::value) {
+      auto& b_adj = forward_as<var_value<matrix_cl<double>>>(b_arena).adj();
+      b_adj = b_adj + res.adj();
+    }
```
Comment (t4c1) [inline stan/math/opencl/rev/add.hpp:46]: If both a and b are vars we would ideally want to only run a single kernel to propagate adjoints.

Comment (t4c1) [reply stan/math/opencl/rev/add.hpp:46]: Same for scalar overload and other functions.

Comment (t4c1) [reply stan/math/opencl/rev/add.hpp:46]: You can use calc_if to make it simpler to implement (see distributions).

**Review by t4c1 — CHANGES_REQUESTED** [review]

File: stan/math/opencl/rev/add.hpp (hunk 1 above)
Comment (rok-cesnovar) [reply stan/math/opencl/rev/add.hpp:50]: Done for non-scalar cases.

**Review by t4c1 — APPROVED** [review]

**Pull request merged**